blocklylib-core/src/main/java/com/google/blockly/android/codegen/LazyServiceConnection.java

```diff
@@ -0,0 +1,71 @@
+package com.google.blockly.android.codegen;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.IBinder;
+
+import com.google.blockly.android.codegen.CodeGeneratorService.CodeGeneratorBinder;
+
+class LazyServiceConnection {
+
+    private final Context mContext;
+    private ServiceConnection mCodeGenerationConnection;
+
+    private CodeGeneratorService mGeneratorService;
+    private CodeGenerationRequest mStoredRequest;
+
+    LazyServiceConnection(Context context) {
+        this.mContext = context;
+    }
+
+    void requestCodeGeneration(CodeGenerationRequest codeGenerationRequest) {
+        if (isBound()) {
+            executeCodeGenerationRequest(codeGenerationRequest);
+        } else {
+            mStoredRequest = codeGenerationRequest;
```

Comment:
Looks like there is a risk of overwriting a prior value of mStoredRequest. To avoid dropping multiple requests, mStoredRequests probably needs to be a queue. That queue may need to be copied to the Service object.

(That said, our callback does not have a mechanism for differentiating multiple requests is the same callback function is used.)